### PR TITLE
setup.py: Require dev release of depman

### PR DESCRIPTION
**Pull request by jayvdb**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -127,6 +128,57 @@ def run(self):
...
+    for i, part in enumerate(version.version):
+        if not isinstance(part, int):
+            break
+
+    previous_final_version = version.version[0:i]
+    previous_final_version[i - 1] -= 1
+
+    version = '.'.join(str(part) for part in previous_final_version)
+
+    return name + '>=' + version
```
Comment (jayvdb) [inline setup.py]: `>` will also work, and be more appropriate.  Still testing that

File: setup.py [hunk 2]
```diff
@@ -127,6 +128,57 @@ def run(self):
...
+    # release in PyPi.
+    # Decrement the first non-final version part, which should be in PyPi,
+    # and use version specifier >= so that the installed package from VCS
+    # will have a version acceptable to the requirement.
+    for i, part in enumerate(version.version):
+        if not isinstance(part, int):
+            break
+
+    previous_final_version = version.version[0:i]
+    previous_final_version[i - 1] -= 1
```
Comment (MacBox7) [inline setup.py]: `coala-utils-5.0.dev0` would result in negative version. Is this a possible scenario?

**Comment by MacBox7** [comment]
`test-requirements.txt` isn't in sync with moban.

**Comment by jayvdb** [comment]
`test-requirements.txt` isnt in scope for this PR.

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**